debugger/src/main/java/org/codice/acdebugger/breakpoints/AccessControlContextCheckProcessor.java

```diff
@@ -47,31 +60,29 @@ public EventRequest createRequest(Debug debug, BreakpointLocation l) throws Exce
   @Override
   public void process(BreakpointInfo info, Debug debug) throws Exception {
     final ThreadReference thread = debug.thread();
+    final ReflectionUtil reflection = debug.reflection();
     final ArrayReference context =
-        debug
-            .reflection()
-            .get(thread.frame(0).thisObject(), "context", "[Ljava/security/ProtectionDomain;");
-    final int local_i = ReadDebugIndex.getIndex(thread);
-    final ObjectReference permission = (ObjectReference) thread.frame(0).getArgumentValues().get(0);
-    final SecurityCheckInformation security =
-        new SecurityCheckInformation(debug, context, local_i, permission);
+        reflection.get(
+            thread.frame(0).thisObject(), "context", "[Ljava/security/ProtectionDomain;");
+    final EnhancedStackFrame frame = enhance(thread.frame(0));
+    final int local_i =
+        reflection.fromMirror(
+            frame.getValue(AccessControlContextCheckProcessor.LOCAL_I_SLOT_INDEX, "I"));
+    final ObjectReference permission = (ObjectReference) frame.getArgumentValues().get(0);
+    final SecurityCheckInformation security = process(debug, context, local_i, permission);
 
     if (security.getFailedDomain() != null) {
-      if (!security.isAcceptable()) {
-        // check if we have only one solution and that solution is to only grant permission(s)
-        // (no privileged blocks) in which case we shall cache them to avoid going through all
-        // of this again
-        final List<SecuritySolution> solutions = security.analyze();
+      // check if we have only one solution and that solution is to only grant permission(s)
```

Comment:
ℹ️ An acceptable failures doesn't have solutions so the code below would be skipped anyway now.